src/report_modules/report_module_json/dependencies/nlohman-json/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 ASAM eV
+#
+# This Source Code Form is subject to the terms of the Mozilla
+# Public License, v. 2.0. If a copy of the MPL was not distributed
+# with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+set(BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
+set(BUILD_TESTS OFF CACHE BOOL "" FORCE)
+set(JSON_BuildTests OFF CACHE INTERNAL "")
+set(JSON_Install OFF CACHE INTERNAL "")
+
+include(FetchContent)
+FetchContent_Declare(nlohmann_json
```

Comment:
Just a remark: Some companies have the constraint about "not downloading at build time". We can consider putting the `json.hpp` header directly in our repo. But as the build of this C++ module can be easily deactivated and we also have the Python version, I'm ok with using `FetchContent` here.

Reply:
I've added an option to disable fetch content. In that case, cmake reverts to an external find_package.

I prefer to avoid "copy-paste" of third party dependencies in a code base. Security issues arises and the deps becomes outdated. More strict policy may be enforced in this case.